src/toolkitAPI.c

```diff
@@ -1248,6 +1248,59 @@ int DLLEXPORT swmm_getSystemRunoffStats(SM_RunoffTotals *runoffTot)
     return(errorcode);
 }
 
+int DLLEXPORT swmm_getGagePrecip(int index, double *value)
+//
+// Input:   index = Index of desired ID
+// Output:  value = value to be output
+// Return:  API Error
+// Purpose: Gets the precipitaion value in the gage. 
+{
+    int errcode = 0;
+    // Check if Open
+    if(swmm_IsOpenFlag() == FALSE)
+    {
+	    errcode = ERR_API_INPUTNOTOPEN;
+    }
+    // Check if object index is within bounds
+    else if (index < 0 || index >= Nobjects[GAGE])
+    {
+	    errcode = ERR_API_OBJECT_INDEX;
+    }
+    // Read the rainfall value
+    else
+    {
+	    *value = Gage[index].rainfall / UCF(RAINFALL);
+    }
+    return(errcode);
+}
+
+int DLLEXPORT swmm_setGagePrecip(int index, double value)
+//
+// Input:   index = Index of desired ID
+//          value = value to be output
+// Return:  API Error
+// Purpose: Sets the precipitation in from the external database
+{
+    int errcode = 0;
+    // Check if Open
+    if(swmm_IsOpenFlag() == FALSE)
+    {
+	    errcode = ERR_API_INPUTNOTOPEN;
+    }
+    // Check if object index is within bounds
+    else if (index < 0 || index >= Nobjects[GAGE])
+    {
+	    errcode = ERR_API_OBJECT_INDEX;
+    }
+    // Read the rainfall value
+    else
+    {
+	    Gage[index].dataSource = RAIN_API;
```

Comment:
add a simple condition here so we do not keep setting it each time it is called:

if (Gage[index].dataSource != RAIN_API) Gage[index].dataSource = RAIN_API;